stdlib/test/collections/test_inline_list.mojo

```diff
@@ -0,0 +1,94 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+# RUN: %mojo %s
+
+from collections import InlineList, Set
+from testing import assert_equal, assert_false, assert_true, assert_raises
+
+
+def test_mojo_issue_698():
+    var list = InlineList[Float64]()
+    for i in range(5):
+        list.append(i)
+
+    assert_equal(0.0, list[0])
+    assert_equal(1.0, list[1])
+    assert_equal(2.0, list[2])
+    assert_equal(3.0, list[3])
+    assert_equal(4.0, list[4])
+
+
+def test_list():
+    var list = InlineList[Int]()
+
+    for i in range(5):
+        list.append(i)
+
+    assert_equal(5, len(list))
+    assert_equal(0, list[0])
+    assert_equal(1, list[1])
+    assert_equal(2, list[2])
+    assert_equal(3, list[3])
+    assert_equal(4, list[4])
+
+    assert_equal(0, list[-5])
+    assert_equal(3, list[-2])
+    assert_equal(4, list[-1])
+
+    list[2] = -2
+    assert_equal(-2, list[2])
+
+    list[-5] = 5
+    assert_equal(5, list[-5])
+    list[-2] = 3
+    assert_equal(3, list[-2])
+    list[-1] = 7
+    assert_equal(7, list[-1])
+
+
+@value
+struct ValueToCountDestructor(CollectionElement):
+    var value: Int
+    var destructor_counter: UnsafePointer[List[Int]]
+
+    fn __del__(owned self):
+        self.destructor_counter[].append(self.value)
+
+
+def test_append_and_destructor():
+    var destructor_counter = List[Int]()
+    var inline_list = InlineList[ValueToCountDestructor, capacity=32]()
+
+    var nb_elements_to_add = 8
+    for index in range(nb_elements_to_add):
+        inline_list.append(
+            ValueToCountDestructor(index, UnsafePointer(destructor_counter))
+        )
+
+    # Using .append() should trigger a move and not a copy+delete.
```

Comment:
👍 would it be easier if we wrote this in terms of `MoveCounter` for counting number of moves/copies for each append call, or is that at the container/`InlineList` level?  See https://github.com/modularml/mojo/blob/668489eee3a48564a453077d42421f290dcb4d14/stdlib/test/test_utils/types.mojo

Reply:
Thanks for the advice, I split this test into two and made the destructor test more robust. Let me know if it's an invalid way to test the destructor